### PR TITLE
Audit report

### PR DESCRIPTION
**Pull request by domludera**
User Stories:  5.1 (#62), 5.2 (#63), 5.3 (#64),  5.6 (#67)

### CODE REVIEW

**Review by NicholasNagy — COMMENTED** [review]
Alright, so under lots of pressures, I realise that a lot of our comments and discussions related to this PR happened offline, so in order to document that, I will include our discussions below:

Firstly, we had issues  with the backend API endpoint for the comments (for US 5.6 #67 ), since we were trying to add a primary key that was unnecessary for comments (as they will never need to be referenced by their primary key), and thus were wasting space. 

The UI of the drill down chart and the UI of the audit report (for US 5.3 #64  ) were also iterated over a bit in order to only show pertinent information useful to understanding the outcome of the audit. Here is the final product, validated by Ajay himself:

image

image

Additionally, we moved the code from a separate branch that needed the code from this one to work effectively (for US 5.2 #63 ) so that we could get immediate feedback from Ajay (for which he mentioned that we will need to change how the report is generated to include the output of actual items found after the audit, not the items from the extract that were used for the audit).

Additionally, changes were asked for the comments, since they originally didn't include the names for each comment box, nor the date and commenting a lot meant having to scroll down to see the newest messages, which wasn't user friendly.

File: server/login/views.py [hunk 1]
```diff
@@ -127,6 +128,10 @@ def save_new_pin(email, user):
     request.data = {'password': first_part + second_part + third_part}
     # For e2e purposes, uncomment this line:
     # request.data = {'password': 'password'}
+    print('\n\n--------------------------')
+    print(request.data)
+    print('--------------------------\n\n')
```
Comment (Amanda-Kolopa) [inline server/login/views.py:133]: Are these required?
If not, please remove it

Comment (NicholasNagy) [reply server/login/views.py:133]: no, we can remove that

**Review by NicholasNagy — APPROVED** [review]
Okay, I looked over the code, including @Amanda-Kolopa 's and I don't see anything new that needs changing. Everything was updated based on our feedback from Ajay, and the user stories are fully completed.

The tests for these changes will be included in a future PR.

For now, it should be good to go. Can you look through my changes @Amanda-Kolopa , so that we can validate each other's code?

**Review by Amanda-Kolopa — APPROVED** [review]
Thank you @NicholasNagy for reviewing the other aspects of the code!
All is working well on my end :smile:

**Pull request merged**